js/pages/characterizations/components/characterizations/characterization-view-edit.js

```diff
@@ -0,0 +1,157 @@
+define([
+    'knockout',
+    'pages/characterizations/services/CharacterizationService',
+    'pages/characterizations/services/PermissionService',
+    'text!./characterization-view-edit.html',
+    'appConfig',
+    'webapi/AuthAPI',
+    'providers/Page',
+    'utils/CommonUtils',
+    'assets/ohdsi.util',
+    'less!./characterization-view-edit.less',
+    'components/tabs',
+    'faceted-datatable',
+    './characterization-view-edit/characterization-design',
+    './characterization-view-edit/characterization-exec-wrapper',
+    './characterization-view-edit/characterization-utils',
+    'components/ac-access-denied',
+], function (
+    ko,
+    CharacterizationService,
+    PermissionService,
+    view,
+    config,
+    authApi,
+    Page,
+    commonUtils,
+    ohdsiUtil
+) {
+    class CharacterizationViewEdit extends Page {
+        constructor(params) {
+            super(params);
+
+            this.selectTab = this.selectTab.bind(this);
+            this.setupDesign = this.setupDesign.bind(this);
+            this.setupSection = this.setupSection.bind(this);
+            this.loadDesignData = this.loadDesignData.bind(this);
+
+            this.characterizationId = ko.observable();
+            this.executionId = ko.observable();
+            this.design = ko.observable({});
+
+            this.designDirtyFlag = ko.observable({ isDirty: () => false });
```

Comment:
design is always pristine?

Reply:
It's re-assigned in `setupDesign()` method

Reply:
You gusy will see in the pathways branch i'm working on (based off this branch) we'll handle the dirtyFlags for these entities differently.  the dirty flag will be stored in the shared state, and so you won't be passing dirty flags around between components: they will all reference the same dirty flag.